clients/src/main/java/org/apache/kafka/clients/producer/ProducerRecord.java

```diff
@@ -54,10 +59,11 @@
      * @param topic The topic the record will be appended to
      * @param partition The partition to which the record should be sent
      * @param timestamp The timestamp of the record
+     * @param headers the headers that will be included in the record
      * @param key The key that will be included in the record
      * @param value The record contents
      */
-    public ProducerRecord(String topic, Integer partition, Long timestamp, K key, V value) {
+    public ProducerRecord(String topic, Integer partition, Long timestamp, Iterable<Header> headers, K key, V value) {
```

Comment:
You use pretty much everywhere K, V, H as parameter order, so would be more consistent to do so here too.